### ISSUE TITLE
Create the Ethereum `state.Database` differently

### ISSUE THREAD

**Issue by joshklop** [polymerdao/monomer#116]
Relevant discussion is here: https://github.com/polymerdao/monomer/pull/114#discussion_r1710278024

The change we need to make is: instead of creating a `state.Database` with `state.NewDatabase`, we should create it with `state.NewDatabaseWithNodeDB`. The latter requires us to pass in a `triedb`, which must be closed.

An alternative is to continue using `state.NewDatabase`, while deferring the call to `TrieDB().Close()`. This could work, but it is less idiomatic and explicit. imho, the fact that the triedb is exposed directly on the `state.Database` via `TrieDB` isn't very nice to begin with.

**Comment by Jesserc** [comment]
Should we do something like this?
`ethstatedb := state.NewDatabaseWithNodeDB(raw, nil)`
Where `nil` replaces a `trieDB` instance

@joshklop

**Comment by joshklop** [comment]
@Jesserc No, the `trieDB` should be initialized like:

```golang
import "github.com/ethereum/go-ethereum/triedb"

trieDB := triedb.NewDatabase(db, nil /* defaults to hash-based scheme */)
env.DeferErr("close trie db", trieDB.Close) // or `t.Cleanup` in tests
```
then we can do
```golang
ethstatedb := state.NewDatabaseWithNodeDB(raw, trieDB)
```

**Comment by Jesserc** [comment]
@joshklop can we close this issue? This was fixed by #114
